pkg/snapshot/restorer/restorer_test.go

```diff
@@ -71,116 +79,522 @@ var _ = Describe("Running Restorer", func() {
+					EmbeddedEtcdQuotaBytes: embeddedEtcdQuotaBytes,
+					BaseSnapshot:           *baseSnapshot,
+					DeltaSnapList:          deltaSnapList,
+				}
+				err = rstr.Restore(RestoreOptions)
+				Expect(err).ShouldNot(HaveOccurred())
+
+			})
 		})
-	})
 
-	Context("with maximum of four fetchers allowed", func() {
-		fmt.Println("Testing for max-fetchers=4")
-		It("should restore etcd data directory", func() {
-			maxFetchers = 4
-
-			restoreOptions := RestoreOptions{
-				ClusterURLs:            clusterUrlsMap,
-				ClusterToken:           restoreClusterToken,
-				RestoreDataDir:         restoreDataDir,
-				PeerURLs:               peerUrls,
-				SkipHashCheck:          skipHashCheck,
-				Name:                   restoreName,
-				MaxFetchers:            maxFetchers,
-				EmbeddedEtcdQuotaBytes: embeddedEtcdQuotaBytes,
-				BaseSnapshot:           *baseSnapshot,
-				DeltaSnapList:          deltaSnapList,
-			}
-			err = rstr.Restore(restoreOptions)
-			Expect(err).ShouldNot(HaveOccurred())
+		Context("with invalid cluster URLS", func() {
+			fmt.Println("Testing for invalid cluster URLS")
+			It("should fail with an error ", func() {
+				restoreCluster = restoreName + "=http://localhost:2390"
+				restorePeerURLs = []string{"http://localhost:2390"}
+				clusterUrlsMap, err = types.NewURLsMap(restoreCluster)
+
+				RestoreOptions := RestoreOptions{
+					ClusterURLs:            clusterUrlsMap,
+					ClusterToken:           restoreClusterToken,
+					RestoreDataDir:         restoreDataDir,
+					PeerURLs:               peerUrls,
+					SkipHashCheck:          skipHashCheck,
+					Name:                   restoreName,
+					MaxFetchers:            maxFetchers,
+					EmbeddedEtcdQuotaBytes: embeddedEtcdQuotaBytes,
+					BaseSnapshot:           *baseSnapshot,
+					DeltaSnapList:          deltaSnapList,
+				}
+				err = rstr.Restore(RestoreOptions)
+				Expect(err).Should(HaveOccurred())
+
+			})
+		})
 
-			err = checkDataConsistency(restoreDataDir, logger)
-			Expect(err).ShouldNot(HaveOccurred())
+		Context("with invalid restore directory", func() {
+			fmt.Println("Testing for invalid restore directory")
+			It("should fail to restore", func() {
+
+				restoreOptions := RestoreOptions{
+					ClusterURLs:            clusterUrlsMap,
+					ClusterToken:           restoreClusterToken,
+					RestoreDataDir:         "", //restoreDataDir,
+					PeerURLs:               peerUrls,
+					SkipHashCheck:          skipHashCheck,
+					Name:                   restoreName,
+					MaxFetchers:            maxFetchers,
+					EmbeddedEtcdQuotaBytes: embeddedEtcdQuotaBytes,
+					BaseSnapshot:           *baseSnapshot,
+					DeltaSnapList:          deltaSnapList,
+				}
+				err = rstr.Restore(restoreOptions)
+				Expect(err).Should(HaveOccurred())
+
+			})
+		})
+
+		Context("with null snapdir and snapname", func() {
+			fmt.Println("Testing for nil snapdir and snapname")
+			It("should not fail and return nil as error", func() {
```

Comment:
Is this a test for the default values for snapdir and snapname?

Reply:
Yes. But as the behavior of restorer is different (restoration can succeed even without base snapshot, given the snapshotter was run without taking a base snapshot), this test would have to be modified so that snapshotter takes only delta snapshots for this test spec. @ashwani2k

Reply:
this test has now been moved under negative test - also a fix is included now in the GetLatestFullSnapshotAndDeltaSnapList() to return sorted the delta snapshots.